woocommerce/woocommerce-gutenberg-products-block
# Refactor BlockTemplateUtils

Slight tidy up of the `BlockTemplateUtils.php` file.

* Moved `remove_theme_templates_with_custom_alternative` from `BlockTemplatesController` to `BlockTemplateUtils`.
* Dropped `gutenberg_` prefix from some methods as this was accidentally carried over from duplicate functions from the Gutenberg codebase.
* Updated some comments around _almost_ duplicate methods.
* Denoted the `BlockTemplateUtils` class as `@internal` to indicate it's not for public use.

### Testing

#### Automated Tests

### Manual Testing

How to test the changes in this Pull Request:

1. Ensure you have a block theme active and WordPress 5.9
2. Load Appearance > Site Editor and check that Woo templates are being loaded in the All Templates view.
3. Load one in the Site Editor and ensure this is loading on the frontend without any customizations
4. Make customizations to the template, and ensure theyre reflected on the frontend.
5. Clear customizations and ensure the frontend template is reset to its original uncustomized state.

Now copy some of the [Woo template files](https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/trunk/templates/templates) into your block theme and repeat steps 1-5 again ensuring your themes templates are overriding the Woo default themes.
